pkg/referrerstore/oras/oras.go

```diff
@@ -375,57 +376,66 @@ func (store *orasStore) GetReferenceManifest(ctx context.Context, subjectReferen
 }
 
 func (store *orasStore) GetSubjectDescriptor(ctx context.Context, subjectReference common.Reference) (*ocispecs.SubjectDescriptor, error) {
-	var desc oci.Descriptor
-	if cachedDesc, ok := store.subjectDescriptorCache.Load(subjectReference.Digest); ok && subjectReference.Digest != "" {
-		desc = cachedDesc.(oci.Descriptor)
-		return &ocispecs.SubjectDescriptor{Descriptor: desc}, nil
-	}
-
-	logrus.Debugf("no digest provided for reference %s. attempting to resolve...", subjectReference.Original)
-	repository, expiry, err := store.createRepository(ctx, store, subjectReference)
+	repository, err := store.createRepository(ctx, store, subjectReference)
 	if err != nil {
 		return nil, err
 	}
 
-	desc, err = repository.Resolve(ctx, subjectReference.Original)
+	desc, err := repository.Resolve(ctx, subjectReference.Original)
 	if err != nil {
 		var ec errcode.Error
 		if errors.As(err, &ec) && (ec.Code == fmt.Sprint(http.StatusForbidden) || ec.Code == fmt.Sprint(http.StatusUnauthorized)) {
-			store.evictAuthCache(subjectReference.Original, err)
+			evict_err := store.evictAuthCache(ctx, subjectReference.Original)
+			if evict_err != nil {
+				err = fmt.Errorf("%s: failed to evict from auth cache: %w", err.Error(), evict_err)
+			}
 		}
 		return nil, err
 	}
-	// add the subject descriptor to cache
-	store.subjectDescriptorCache.Store(desc.Digest, desc)
-	// add the repository client to the auth cache if all repository operations successful
-	store.addAuthCache(subjectReference.Original, repository, expiry)
 
 	return &ocispecs.SubjectDescriptor{Descriptor: desc}, nil
 }
 
-func createDefaultRepository(ctx context.Context, store *orasStore, targetRef common.Reference) (registry.Repository, time.Time, error) {
+func createDefaultRepository(ctx context.Context, store *orasStore, targetRef common.Reference) (registry.Repository, error) {
 	if store.authProvider == nil || !store.authProvider.Enabled(ctx) {
-		return nil, time.Now(), fmt.Errorf("auth provider not properly enabled")
+		return nil, fmt.Errorf("auth provider not properly enabled")
 	}
-
-	if entry, ok := store.authCache.Load(targetRef.Original); ok {
-		// if the auth cache entry expiration has not expired or it was never set
-		cacheEntry := entry.(authCacheEntry)
-		if cacheEntry.expiresOn.IsZero() || cacheEntry.expiresOn.After(time.Now()) {
-			return cacheEntry.client, cacheEntry.expiresOn, nil
-		}
-	}
-
-	authConfig, err := store.authProvider.Provide(ctx, targetRef.Original)
+	artifactRef, err := registry.ParseReference(targetRef.Original)
 	if err != nil {
-		logrus.Warningf("auth provider failed with err, %v", err)
-		logrus.Info("attempting to use anonymous credentials")
+		return nil, err
+	}
+	var authConfig authprovider.AuthConfig
+	cacheProvider := cache.GetCacheProvider()
+	var cacheResponse string
+	found := false
+	if cacheProvider != nil {
```

Comment:
We should consider security boundary and if it's safe to store registry creds inside the cache without encrypting.